contracts/truefi2/LineOfCreditAgency.sol

```diff
@@ -369,8 +375,10 @@ contract LineOfCreditAgency is UpgradeableClaimable, ILineOfCreditAgency {
             "LineOfCreditAgency: Sender not eligible to borrow"
         );
         require(!_hasOverdueInterest(pool, msg.sender), "LineOfCreditAgency: Sender has overdue interest in this pool");
-        (uint8 oldScore, uint8 newScore) = _updateCreditScore(pool, msg.sender);
-        require(newScore >= minCreditScore, "LineOfCreditAgency: Borrower has credit score below minimum");
+        uint256 currentPrincipal = borrowed[pool][msg.sender];
+        uint256 newPrincipal = currentPrincipal.add(amount);
+        (uint8 oldEffectiveScore, uint8 newEffectiveScore) = _updateCreditScore(pool, msg.sender, newPrincipal);
+        require(newEffectiveScore >= minCreditScore, "LineOfCreditAgency: Borrower has credit score below minimum");
```

Comment:
Can we add a test for a situation where borrower couldn't borrow due to their low credit score, but after staking some TRU they could?

Reply:
Let's do that in another PR. Task filed.